types/react/index.d.ts

```diff
@@ -897,18 +893,11 @@ declare namespace React {
 
     // I'm not sure if I keep this 2-ary or if I make it (2,3)-ary; it's currently (2,3)-ary.
     // The Flow types do have an overload for 3-ary invocation with undefined initializer.
-
-    // NOTE: without the ReducerState indirection, TypeScript would reduce S to be the most common
```

Comment:
This note makes it sound like that the previous behavior was intentional. Could you do a git blame and ping the person that added this? I think this is not about type checking but autocompletion only which has to be tested manuall.y

Reply:
It seems that, while autocomplete _does_ still work, the behavior is... strange.

Here is autocomplete working; note the inferred S generic:
![image](https://user-images.githubusercontent.com/73085/60577939-4c88f500-9dbb-11e9-897c-f071739645d3.png)

But note how it is inferred after you enter the initializer:

![image](https://user-images.githubusercontent.com/73085/60578026-77734900-9dbb-11e9-860d-d2dc9b635829.png)

---

Like, it's not **wrong**, but I wonder why does the object prevent the type alias from being used? The generic is invariant here (you can't either omit or add keys), but typescript seems to still construct a singleton common subtype instead of reusing the alias.

Reply:
I see the difference:

**current behavior in master**
<img width="949" alt="Screen Shot 2019-07-03 at 11 46 22 AM" src="https://user-images.githubusercontent.com/25131/60617554-d53a6d80-9d88-11e9-88f4-bef7707b3bdf.png">

**new behavior in my branch**
<img width="632" alt="Screen Shot 2019-07-03 at 11 48 20 AM" src="https://user-images.githubusercontent.com/25131/60617626-f438ff80-9d88-11e9-8f0c-e6d0bd8cdff3.png">